src/utils/blockly-api.js

```diff
@@ -91,6 +83,14 @@ class BlocklyApi {
     interpreter.setProperty(scope, 'getLightSensorValue',
       interpreter.createNativeFunction(wrapper));
 
+    // Add get button pressed API function
+    wrapper = (sensor) => {
+      const buttonReading = this.sensorStateCache[`${sensor}_BUTTON`];
+      return interpreter.createPrimitive(buttonReading);
```

Comment:
I'm not seeing the indicator get cleared after checking the status. I think we'd need to do that here?

Reply:
Disregard. We're gonna go with the toggle method.